Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1125,6 +1125,8 @@ Ref<WebPageProxy> WebProcessPool::createWebPage(PageClient& pageClient, Ref<API:
 
     auto page = process->createWebPage(pageClient, WTFMove(pageConfiguration));
 
+    page->websiteDataStore().networkProcess().setBlobRegistryTopOriginPartitioningEnabled(page->websiteDataStore().sessionID(), page->preferences().store().getBoolValueForKey(WebPreferencesKey::blobRegistryTopOriginPartitioningEnabledKey()));
```

Comment:
This isn't great. If you create one page with the setting on, then another with the setting off, you end up with partitioning disabled.

Ideally, we'd turn it on as soon as one page requires it and turn it off as soon as no page requires it.
We could have an `updateBlobRegistryPartitioningState()` function that'd we'd call every time a page gets added or removed. This function would iterate through all pages and see if any page needs the setting to be on.

Reply:
I was more concerned with testing, and so immediately enabling/disabling partitioning based on the preference seemed like the most obvious behavior, but I'm fine with making it behave like you described.